packages/java/parser-jvm-plugin-backbone/src/main/java/dev/hilla/parser/plugins/backbone/PropertyPlugin.java

```diff
@@ -0,0 +1,194 @@
+import dev.hilla.parser.models.jackson.JacksonPropertyModel;
+import dev.hilla.parser.plugins.backbone.nodes.EntityNode;
+import dev.hilla.parser.plugins.backbone.nodes.PropertyNode;
+
+import jakarta.annotation.Nonnull;
+
+public final class PropertyPlugin extends AbstractPlugin<PluginConfiguration> {
+    private static final Comparator<BeanPropertyDefinition> sorter = Comparator
+            .comparing(BeanPropertyDefinition::getName);
+    private final SerializationConfig config;
+
+    public PropertyPlugin(@Nonnull ObjectMapper mapper) {
+        this.config = Objects.requireNonNull(mapper).getSerializationConfig();
+    }
+
+    @Override
+    public void enter(NodePath<?> nodePath) {
+        if (nodePath.getNode() instanceof PropertyNode) {
+            var propertyNode = (PropertyNode) nodePath.getNode();
+            propertyNode.setTarget(propertyNode.getSource().getName());
+        }
+    }
+
+    @Override
+    public void exit(NodePath<?> nodePath) {
+    }
+
+    @Nonnull
+    @Override
+    public NodeDependencies scan(@Nonnull NodeDependencies nodeDependencies) {
+        if (!(nodeDependencies.getNode() instanceof EntityNode)) {
+            return nodeDependencies;
+        }
+
+        var model = (ClassInfoModel) nodeDependencies.getNode().getSource();
+        if (model.isEnum()) {
+            return nodeDependencies;
+        }
+
+        var properties = collectProperties(model)
+                .<Node<?, ?>> map(PropertyNode::of);
+
+        return nodeDependencies.appendChildNodes(properties);
+    }
+
+    private Stream<JacksonPropertyModel> collectProperties(
+            @Nonnull ClassInfoModel model) {
+        var cls = Objects.requireNonNull(model).get();
+
+        if (!(cls instanceof Class<?>)) {
+            throw new BackbonePluginException(
+                    "Jackson: Only reflection models are supported");
+        }
+
+        var description = config
+                .introspect(config.constructType((Class<?>) cls));
+
+        var processor = new PropertyProcessor(description);
+        return processor.stream().sorted(sorter).map(JacksonPropertyModel::of);
+    }
+
+    /**
+     * The class for processing each BeanDescription. The class' algorithm is
+     * adopted from the Jackson's BeanSerializerFactory class (v2.15).
+     */
+    private class PropertyProcessor {
+        private final BeanDescription description;
+
+        PropertyProcessor(BeanDescription description) {
+            this.description = description;
+        }
+
+        public Stream<BeanPropertyDefinition> stream() {
+            var properties = description.findProperties().stream();
+            properties = filterSuperClassProperties(properties);
+            properties = filterPropertiesWithIgnoredTypes(properties);
+
+            // Remove ones without matching mutator (if it is configured).
+            if (config.isEnabled(MapperFeature.REQUIRE_SETTERS_FOR_GETTERS)) {
```

Comment:
What's the use case for `REQUIRE_SETTERS_FOR_GETTERS`?

Reply:
Jackson [has](https://github.com/FasterXML/jackson-databind/blob/2.15/src/main/java/com/fasterxml/jackson/databind/ser/BeanSerializerFactory.java#L606) it in the source code. The whole class is just a adapted code from the Jackson BeanSerializationFactory. I guess all of it is necessary for the same behavior in generator and controller.